### PR TITLE
New ResizableWidget is a formal way to request layouts

### PR DESCRIPTION
**Pull request by swankjesse**
This is only on our UIView APIs because that's the only view system that needs an out-of-band mechanism to signal such changes.

This also introduces new optional parameters to UIViewRedwoodLayoutWidgetFactory and ViewRedwoodLayoutWidgetFactory to opt-in to experimental incremental layouts. This mechanism is still a work-in-progress so most users shouldn't opt in yet.

Working towards: https://github.com/cashapp/redwood/issues/2291

---

- [x] `CHANGELOG.md`'s "Unreleased" section has been updated, if applicable.

### CODE REVIEW

File: redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractFlexContainerTest.kt [hunk 1]
```diff
@@ -625,6 +626,42 @@ abstract class AbstractFlexContainerTest<T : Any> {
...
+      .apply { modifier = HeightImpl(100.dp) }
+      .also { container.add(it) }
+    val c = text("C")
+      .apply { modifier = HeightImpl(100.dp) }
+      .also { container.add(it) }
+    container.onEndChanges()
+    verifySnapshot(container.value, "initial")
+
+    val aMeasureCount = a.measureCount
+    val bMeasureCount = a.measureCount
```
Comment (ChristianKatzmann) [inline redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractFlexContainerTest.kt]:
```suggestion
    val bMeasureCount = b.measureCount
```

Comment (swankjesse) [reply redwood-layout-shared-test/src/commonMain/kotlin/app/cash/redwood/layout/AbstractFlexContainerTest.kt]: Thanks!

File: redwood-lazylayout-uiview/RedwoodLazylayoutUIViewTests/__Snapshots__/UIViewLazyListAsFlexContainerTestHost/testRecursiveLayoutIsIncremental.v1.png
Comment (JakeWharton) [inline redwood-lazylayout-uiview/RedwoodLazylayoutUIViewTests/__Snapshots__/UIViewLazyListAsFlexContainerTestHost/testRecursiveLayoutIsIncremental.v1.png:1]: What's going on here? Broken?

File: redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/YogaUIView.kt [hunk 2]
```diff
@@ -111,7 +112,11 @@ internal class YogaUIView(
       applyModifier(node, index)
     }
 
-    size.useContents { rootNode.measure(width.toFloat(), height.toFloat()) }
+    if (!incremental) {
+      size.useContents { rootNode.markEverythingDirty() }
```
Comment (JakeWharton) [inline redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/YogaUIView.kt]:
```suggestion
      rootNode.markEverythingDirty()
```

**Review by JakeWharton — APPROVED** [review]
I can't believe this responsibility is placed on the user in UIKit...

File: redwood-lazylayout-uiview/RedwoodLazylayoutUIViewTests/__Snapshots__/UIViewLazyListAsFlexContainerTestHost/testRecursiveLayoutIsIncremental.v1.png
Comment (swankjesse) [reply redwood-lazylayout-uiview/RedwoodLazylayoutUIViewTests/__Snapshots__/UIViewLazyListAsFlexContainerTestHost/testRecursiveLayoutIsIncremental.v1.png:1]: Yeah, lemme dig into this one next.

Comment (swankjesse) [reply redwood-lazylayout-uiview/RedwoodLazylayoutUIViewTests/__Snapshots__/UIViewLazyListAsFlexContainerTestHost/testRecursiveLayoutIsIncremental.v1.png:1]: The weird part is it’s V1 so it shouldn’t be _this_ broken.

**Pull request merged**